packages/intrepid2/src/Shared/Intrepid2_Data.hpp

```diff
@@ -0,0 +1,1843 @@
+//
+//  Intrepid2_Data.hpp
+//  QuadraturePerformance
+//
+//  Created by Roberts, Nathan V on 8/24/20.
+//
+
+#ifndef Intrepid2_Data_h
+#define Intrepid2_Data_h
+
+#include "Intrepid2_ScalarView.hpp"
+#include "Intrepid2_Utils.hpp"
+
+/** \file  Intrepid2_Data.hpp
+   \brief  Defines the Data class, a wrapper around a Kokkos::View that allows data that is constant or repeating in various notional dimensions to be stored just once, while providing a similar interface to that of View.
+   \author Created by N.V. Roberts.
+*/
+
+namespace Intrepid2 {
+/** \enum  Intrepid2::DataVariationType
+    \brief Enumeration to indicate how data varies in a particular dimension of an Intrepid2::Data object.
+*/
+  enum DataVariationType
```

Comment:
@mperego, thanks for the suggestion.  I've edited so that the Doxygen now reads:

```
/** \enum  Intrepid2::DataVariationType
    \brief Enumeration to indicate how data varies in a particular dimension of an Intrepid2::Data object.
 CONSTANT indicates that the data does not vary; MODULAR indicates that it varies according to some (separately specified) modulus; BLOCK_PLUS_DIAGONAL allows specification of a matrix that has a non-diagonal block followed by a diagonal block; GENERAL indicates arbitrary variation.
 
 To give some examples for a Data object containing the Jacobian for reference-to-physical space mappings:
 - CONSTANT could be used in the point dimension for an affine transformation
 - MODULAR could be used for the cell dimension for a uniform grid that has been subdivided into simplices
 - BLOCK_PLUS_DIAGONAL could be used for the coordinate dimensions for an arbitrary 2D mesh that has been orthogonally extruded in the z dimension (resulting in diagonal entries in the final row and column of the Jacobian matrix)
 - GENERAL should be used in any dimension in which the data varies in a way not captured by the other options
*/
```